### PR TITLE
Add support for ruby >= 2.3 and rails 5.x

### PR DESCRIPTION
**Pull request by maeve**
Bump the major version because we've also dropped support for ruby < 2.2 and rails < 4.1.

* Depends on:
  * G5/g5_authentication_client#33
  * G5/devise_g5_authenticatable#27
* Required for  [#144943067](https://www.pivotaltracker.com/story/show/144943067)

### CODE REVIEW

**Comment by maeve** [comment]
This already went through QA as part of rails 5 upgrades to various apps.

File: lib/g5_authenticatable_api/services/token_info.rb [hunk 1]
```diff
@@ -1,35 +1,51 @@
...
       end
 
       private
+
       def extract_token_from_header
-        if authorization_header
-          parts = authorization_header.match(/Bearer (?<access_token>\S+)/)
-          parts['access_token']
-        end
+        return unless authorization_header
```
Comment (perryqh) [inline lib/g5_authenticatable_api/services/token_info.rb]: You've come over to the dark side, too. Early-returns are an anti-pattern in many other languages.

**Review by perryqh — APPROVED** [review]
💯

File: lib/g5_authenticatable_api/services/token_info.rb (hunk 1 above)
Comment (maeve) [reply lib/g5_authenticatable_api/services/token_info.rb]: Indeed. The first time rubocop suggested an early return with a guard clause, it felt so...wrong. I resisted for a long time. Even now, I think twice before I use it. But in those cases where you really do need some defensive programming (like when you're inspecting a request that could've come from anywhere), it makes the code much cleaner.

File: spec/dummy/config/environments/test.rb [hunk 2]
```diff
@@ -13,8 +15,22 @@
   config.eager_load = false
 
   # Configure static asset server for tests with Cache-Control for performance.
-  config.serve_static_assets  = true
-  config.static_cache_control = "public, max-age=3600"
+  cache_header_value = 'public, max-age=3600'
+
+  if config.respond_to?(:public_file_server)
```
Comment (dpetersen) [inline spec/dummy/config/environments/test.rb:20]: If this is a Rails 4/5 thing, I think you should comment which is which for your future sanity and ripping out support for 4.

Comment (dpetersen) [reply spec/dummy/config/environments/test.rb:20]: oh nevermind this is spec/dummy, I don't care that much

**Comment by dpetersen** [comment]
Looks good. 👍

**Pull request merged**